app/lib/validator_manager.py

```diff
@@ -56,34 +56,44 @@ def get_validators_pending_deletion(ws_endpoint):
         return list(set(active_validators) - set(staking_validators))
 
 
-def setup_pos_validator(ws_endpoint, session_key,  stash_seed, controller_seed=None):
+def setup_pos_validator(ws_endpoint, stash_seed, session_key=None, controller_seed=None):
     batch_call = []
     substrate_client = get_substrate_client(ws_endpoint)
     stash_keypair = Keypair.create_from_uri(stash_seed)
 
-    batch_call.append(substrate_client.compose_call(
-            call_module='Staking',
-            call_function='bond',
-            call_params={
-                'controller': stash_keypair.ss58_address,
-                'value': 1 * 10 ** 11,
-                'payee': "Staked"
-            }
+    # 1. Bond
+    bonded = substrate_client.query('Staking', 'Bonded', params=[stash_keypair.ss58_address]).value
+    # If we already bond, skip this step
+    if not bonded or not (bonded in [stash_keypair.ss58_address, controller_seed]):
+        batch_call.append(
+            substrate_client.compose_call(
+                call_module='Staking',
+                call_function='bond',
+                call_params={
+                    'controller': stash_keypair.ss58_address,
+                    'value': 1 * 10 ** 11,
+                    'payee': "Staked"
+                }
+            )
         )
-    )
 
-    if type(session_key) == str:
-        session_key = decode_session_key(substrate_client,session_key)
-    batch_call.append(substrate_client.compose_call(
-            call_module='Session',
-            call_function='set_keys',
-            call_params={
-                'keys': session_key,
-                'proof': ''
-            }
+    # 2. Set session key. If a session key is not provided we assume what session key is already set ( Example: we are
+    # registering deregistered validator)
+    if session_key:
+        if type(session_key) == str:
```

Comment:
Not really, first "if" checks if the `session key` is not "None" and the second "if" checks the type and if it is the string  it will convert to dict